### PR TITLE
.github/workflows/integration.yaml: Take gosec into use

### PR DESCRIPTION
**Pull request by maniankara**
Have gosec, snyk , unit tests and integration tests running in our GitHub actions workflows and results captured with Bubbly into generic tables. 

Quote from Jacob:
> E.g. we don't want snyk_vulnerability  but instead something like table "cve" { ... } , and then it doesn't matter if you run snyk or some other tool and the goal of the pipeline is to standardise all the results so that we can use them for "release readiness"

### CODE REVIEW

**Comment by maniankara** [comment]
Stop failing on error and output to json

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -17,6 +17,8 @@ jobs:
         go-version: [1.16.2]
         os: [ubuntu-latest]
     runs-on: ${{ matrix.os }}
+    env:
+      GO111MODULE: on
```
Comment (olliefr) [inline .github/workflows/integration.yaml:21]: I'm curious about this. We request Go 1.16 in line 17, and `GO111MODULE` is `on` by default in 1.16. What am I missing?

**Review by olliefr — COMMENTED** [review]
I'm a bit out of the loop on this. What's the purpose of this PR? Is it to stop the commits which may introduce known security vulnerabilities into the `main`? Does this Action actually stop such commits from being merged into `main` or does it serve only as a warning?

File: .github/workflows/integration.yaml (hunk 1 above)
Comment (maniankara) [reply .github/workflows/integration.yaml:21]: I should remove it, please hold on :) I will finish it and let you guys know.

**Comment by maniankara** [comment]
Hello guys, now this can be discussed to be reviewed. This was done:
* Run gosec and get the output as json so that it can be imported to bubbly
* Convert the integration test results to json for the same reason above.

**Comment by maniankara** [comment]
Should I add snyk to the same PR ? 
@olliefr wanted to check also go lint, should this be added here as well?
About unit tests, they are run on a different job, so does not have access to bubbly.

**Comment by olliefr** [comment]
With regards to `golint`, it would be a separate PR and we had not made a decision on that yet. There are some existing issues to be fixed before this can be done.

File: .github/workflows/integration.yaml [hunk 2]
```diff
@@ -38,4 +38,9 @@ jobs:
           # just sleep a bit to wait for the containers to start
           sleep 1
           docker-compose logs
-          go test ./... -tags=integration -count=1
+          go test ./... -tags=integration -count=1 -json | tee integration.json
+      - name: Run Gosec Security Scanner
+        run: |
+          curl -sfL https://raw.githubusercontent.com/securego/gosec/master/install.sh | sh -s -- -b $(go env GOPATH)/bin latest
+          gosec -fmt json -no-fail -out out.json ./...
+          head -n 30 out.json # to be removed
```
Comment (olliefr) [inline .github/workflows/integration.yaml:46]: Is this line supposed to be here now that the PR is final? 🙃

Comment (maniankara) [reply .github/workflows/integration.yaml]: My plan was to attach this PR to also upload this gosec json into local bubbly (running out of `docker-compose`). Apparently, thats going to take time  :) So removing the `todo`

**Review by jlarfors — APPROVED** [review]
LGTM! I would consider setting up golangci-lint in the future? It seems quite convenient if we want to run a number of different tools

https://golangci-lint.run/

**Review by olliefr — APPROVED** [review]

**Comment by olliefr** [comment]
@maniankara and @jlarfors I would suggest making a separate PR for `golangci-lint`, or creating an issue for it.

**Pull request merged**